fontir/src/ir.rs

```diff
@@ -115,41 +116,122 @@ pub struct GlyphInstance {
     /// Advance height; if None, assumed to equal font's ascender - descende.
     pub height: Option<f64>,
     /// List of glyph contours.
-    pub contours: Vec<Contour>,
+    pub contours: Vec<BezPath>,
     /// List of glyph components.
     pub components: Vec<Component>,
 }
 
-/// A single glyph contour consisting of a list of points.
-pub type Contour = Vec<ContourPoint>;
-
-/// A single point in a glyph contour.
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
-pub struct ContourPoint {
-    pub x: f64,
-    pub y: f64,
-    pub typ: PointType,
-}
-
-/// Possible types of a point in a glyph contour, following UFO GLIF semantics.
-#[derive(Serialize, Deserialize, Debug, Clone, PartialEq, Eq)]
-pub enum PointType {
-    Move,
-    Line,
-    OffCurve,
-    Curve,
-    QCurve,
-}
-
 /// A single glyph component, reference to another glyph.
 #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
 pub struct Component {
     /// The name of the referenced glyph.
-    pub base: String,
+    pub base: GlyphName,
     /// Affine transformation to apply to the referenced glyph.
     pub transform: Affine,
 }
 
+/// Helps convert points-of-type to a bezier path.
+///
+/// Source formats tend to use streams of point-of-type. Curve manipulation is
+/// often easier on bezier path, so provide a mechanism to convert.
+#[derive(Debug)]
+pub struct GlyphPathBuilder<'a> {
+    glyph_name: &'a str,
```

Comment:
we have a generally cheap `GlyphName` now, would it be easier to use use that and then not worry about the lifetime?

Reply:
Yes, it would indeed. Ty, will fix.